### PR TITLE
Feat/wallet get capabilities cache

### PR DESCRIPTION
**Pull request by ganchoradkov**
## Description
Implemented `wallet_getCapabilities` caching in `universal-provider`

## Type of change

- [ ] Chore (non-breaking change that addresses non-functional tasks, maintenance, or code quality improvements)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Draft PR (breaking/non-breaking change which needs more work for having a proper functionality _[Mark this PR as ready to review only when completely ready]_)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How has this been tested?
tests 

## Checklist

- [ ] I have performed a self-review of my own code
- [ ] My changes generate no new warnings
- [ ] Any dependent changes have been merged and published in downstream modules

# Additional Information (Optional)

Please include any additional information that may be useful for the reviewer.

### CODE REVIEW

File: providers/universal-provider/src/providers/eip155.ts [hunk 1]
```diff
@@ -168,6 +170,34 @@ class Eip155Provider implements IProvider {
...
+    const capabilities = await this.client.request(args as EngineTypes.RequestParams);
+    try {
+      // update the session with the capabilities so they can be retrieved later
+      await this.client.session.update(args.topic, {
+        sessionProperties: {
+          ...(session.sessionProperties || {}),
+          capabilities: {
+            ...(sessionCapabilities || {}),
+            [address]: capabilities,
+          } as any, // by spec sessionProperties should be <string, string> but here are used as objects?
```
Comment (arein) [inline providers/universal-provider/src/providers/eip155.ts:193]: what do we do about this? The `spec` is the EIP-5792 spec. I think we do something else maybe in our CAIP-25 spec? So maybe we just cover the "right" think in our wallet docs / wallet SDK?

We probably need to expand our namespace builder for this usecase

File: providers/universal-provider/test/index.spec.ts [hunk 2]
```diff
@@ -1001,6 +1002,78 @@ describe("UniversalProvider", function () {
           expectedChainId,
         });
       });
+      it("should cache `wallet_getCapabilities` request", async () => {
```
Comment (arein) [inline providers/universal-provider/test/index.spec.ts:1005]: do you need a negative test here if not cached?

**Review by arein — APPROVED** [review]

File: providers/universal-provider/src/providers/eip155.ts (hunk 1 above)
Comment (ganchoradkov) [reply providers/universal-provider/src/providers/eip155.ts:193]: according to `caip-25` spec the sessionProperties (scopeProperties) is `<string, string>`

File: providers/universal-provider/test/index.spec.ts (hunk 2 above)
Comment (ganchoradkov) [reply providers/universal-provider/test/index.spec.ts:1005]: the first part of the test where the wallet receives the request covers the not cached part. If it was cached already the request wouldn't have been sent to the wallet

**Review by tomiir — APPROVED** [review]

**Pull request merged**